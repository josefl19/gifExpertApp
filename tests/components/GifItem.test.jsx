const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe('Test para el componente GifItem', () => {
    const title = 'Titulo del item';
    const category = 'Dogsu'
    const url = `https://api.giphy.com/v1/gifs/search?api_key=FtiAvZc9zt3N8G5MwZkwoJBFdJizTyEY&q=${category}&limit=10`;

    test('Hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y el ATL indicado', () => {
        render(<GifItem title={title} url={url} />)

        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});