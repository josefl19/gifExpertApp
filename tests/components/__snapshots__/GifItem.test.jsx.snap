// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el componente GifItem Hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo del item"
      src="https://api.giphy.com/v1/gifs/search?api_key=FtiAvZc9zt3N8G5MwZkwoJBFdJizTyEY&q=Dogsu&limit=10"
    />
    <p>
      Titulo del item
    </p>
  </div>
</div>
`;
